From b2f40b03ce1778c7e97586e4111d22d3b28b3330 Mon Sep 17 00:00:00 2001
From: Artem Savkov <asavkov@redhat.com>
Date: Fri, 7 Dec 2018 16:46:08 +0100
Subject: [PATCH] kmod/patch: more linking fixes

While adding proper linker script option my previous patch left the
linker script in the list of source files (on pre-4.20 kernels) for
ld somehow breaking kpatch callback sections. For this to work
properly kpatch.lds needs to be added to 'extra-y' instead of objs. And
for kbuild to process this option properly we need to call make without
the .ko target, i.e. let kbuild decide what to build.

Fixes: 17a97b4 ("kmod/patch: fix patch linking with 4.20")
Signed-off-by: Artem Savkov <asavkov@redhat.com>
---
 kmod/patch/Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/kmod/patch/Makefile b/kmod/patch/Makefile
index 7ba0820..e017b17 100644
--- a/kmod/patch/Makefile
+++ b/kmod/patch/Makefile
@@ -12,13 +12,14 @@ endif
 
 obj-m += $(KPATCH_NAME).o
 ldflags-y += -T $(src)/kpatch.lds
+extra-y := kpatch.lds
 
-$(KPATCH_NAME)-objs += patch-hook.o kpatch.lds output.o
+$(KPATCH_NAME)-objs += patch-hook.o output.o
 
 all: $(KPATCH_NAME).ko
 
 $(KPATCH_NAME).ko:
-	$(KPATCH_MAKE) $(KPATCH_NAME).ko
+	$(KPATCH_MAKE)
 
 patch-hook.o: patch-hook.c kpatch-patch-hook.c livepatch-patch-hook.c
 	$(KPATCH_MAKE) patch-hook.o
-- 
1.7.12.4

